libarchive_fe/passphrase.c

```diff
@@ -113,8 +114,7 @@ readpassphrase(const char *prompt, char *buf, size_t bufsiz, int flags)
 	WriteFile(hStdout, "\r\n", 2, NULL, NULL);
 	buf[rbytes] = '\0';
 	/* Remove trailing carriage return(s). */
-	if (rbytes > 2 && buf[rbytes - 2] == '\r' && buf[rbytes - 1] == '\n')
-		buf[rbytes - 2] = '\0';
+	buf[strcspn(buf, "\r\n")] = '\0';
```

Comment:
I might be miscalibrated here, but I am worried about introducing another uncontrolled null-terminated-string libc API here. I admit that we did just make it safe on line R115...

Reply:
I'd also like to know why the previous code failed.  (Mostly just to satisfy my curiosity.)

Reply:
The bug is very subtle: The first condition in the if-condition states `rbytes > 2`. If you press enter directly, exactly 2 bytes are read: `\r\n`. This implies that the if-condition cannot be true, so `\r\n` is not stripped.

A simpler solution could be to write `rbytes > 1` instead or `rbytes >= 2`. I decided to use `strcspn` instead to get closer to the functionality of the original `readpassphrase` POSIX version in this file.

There are still differences, mostly that `ReadFile` could probably read more than just one line, but I am not sure about that. I am not an expert when it comes to Windows API. Also the no-echo mode of stdin is not reset. Since I am not sure if this could ever be noticed with any real world scenario, I did not adjust these things, yet.

Reply:
Ah.  That makes sense.  Thanks for the clarification.